scripts/script.js

```diff
@@ -0,0 +1,186 @@
+const container = document.querySelector(".container");
+const display = document.getElementById("display");
+const historyList = document.getElementById("history");
+const memoryList = document.getElementById("memory");
+
+let memory = 0;
+let history = [];
+let string = "";
+
+container.addEventListener("click", (e) => {
+  const button = e.target;
+
+  const value = button.value;
+
+  if (value === "C") {
+    display.value = "0";
+  } else if (value === "BACK") {
+    display.value = display.value.slice(0, -1);
+  } else if (value === "=") {
+    let result = display.value;
+    if (result == "") {
+      return;
+    }
+    if (result.includes("ln")) {
+      if (result.includes("ln(")) {
+        const number = parseFloat(
+          result.replace(/ln\(([^)]+)\)/g, (match, p1) => {
+            return Math.log(parseFloat(p1));
+          })
+        );
+        display.value = number;
+        return;
+      }
+    }
+
+    if (result.includes("log")) {
+      const number = parseFloat(
+        result.replace(/log\(([^)]+)\)/g, (match, p1) => {
+          return Math.log10(parseFloat(p1));
+        })
+      );
+      display.value = number;
+      return;
+    }
+
+    if (result.includes("sin")) {
+      const number = parseFloat(
+        result.replace(/sin\(([^)]+)\)/g, (match, p1) => {
+          return Math.sin(p1);
+        })
+      );
+      display.value = number;
+      return;
+    }
+
+    if (result.includes("cos")) {
+      const number = parseFloat(
+        result.replace(/cos\(([^)]+)\)/g, (match, p1) => {
+          return Math.cos(p1);
+        })
+      );
+      display.value = number;
+      return;
+    }
+
+    if (result.includes("tan")) {
+      const number = parseFloat(
+        result.replace(/tan\(([^)]+)\)/g, (match, p1) => {
+          return Math.tan(p1);
+        })
+      );
+      display.value = number;
+      return;
+    }
+
+    if (result.includes("!")) {
+      const number = parseInt(result.replace("!", ""));
+      display.value = factorial(number);
+      return;
+    }
+
+    if (result.includes("2√")) {
+      display.value = Math.sqrt(parseFloat(result.replace("2√", "")));
+      return;
+    }
+
+    if (result.includes("^")) {
+      const [base, exponent] = result.split("^");
+      display.value = exponentiate(parseFloat(base), parseFloat(exponent));
+      return;
+    }
+
+    try {
+      let evaluatedResult = eval(result);
+      display.value = evaluatedResult;
+      history.push(`${result} = ${evaluatedResult}`);
+      updateHistory();
+    } catch (error) {
+      display.value = "Error";
+    }
+  } else if (value === "Pi") {
+    display.value =
+      display.value === "" ? Math.PI : parseFloat(display.value) * Math.PI;
```

Comment:
create a displayValue variable instead of repeating this same everywhere

Reply:
okay